week05/errors_test.go

```diff
@@ -0,0 +1,182 @@
+package week05
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestErrTableNotFound_TableNotFound(t *testing.T) {
+
+	t.Parallel()
+
+	testcases := []struct {
+		name string
+		e    ErrTableNotFound
+		want string
+	}{
+		{"empty ErrTableNotFound", ErrTableNotFound{}, ""},
+		{"ErrTableNotFound with correct table", ErrTableNotFound{table: "users"}, "users"},
+		{"ErrTableNotFound with wrong table", ErrTableNotFound{table: "people"}, "users"},
+	}
+
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+
+			got := tt.e.TableNotFound()
+
+			if got != tt.want {
+				t.Fatalf("unexpected error: got %v, want %v", got, tt.want)
+			}
+
+		})
+	}
+}
+
+func TestErrTableNotFound_IsErrTableNotFound(t *testing.T) {
+
+	t.Parallel()
+
+	testcases := []struct {
+		name string
+		err  error
+		want bool
+	}{
+		{"not ErrTableNotFound error", fmt.Errorf("hello"), false},
+		{"not not ErrTableNotFound error", &ErrTableNotFound{table: "users"}, true},
+		{"empty ErrTableNotFound error", &ErrTableNotFound{}, true},
+	}
+
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+
+			got := IsErrTableNotFound(tt.err)
+
+			if got != tt.want {
+				t.Fatalf("unexpected error: got %v, want %v", got, tt.want)
+			}
+
+		})
+	}
+}
+
+func TestErrNoRows_Clauses(t *testing.T) {
+
+	t.Parallel()
+
+	testcases := []struct {
+		name string
+		e    *errNoRows
+		want Clauses
+	}{
+		{"empty clauses provided", &errNoRows{Clauses{}, "users"}, Clauses{"hobby": "golf", "name": "Webbs"}},
+		{"clauses provided correct expected", &errNoRows{Clauses{"hobby": "golf", "name": "Webbs"}, "users"}, Clauses{"name": "Webbs", "hobby": "golf"}},
+		{"clauses provided wrong expected", &errNoRows{Clauses{"name": "Lion", "continent": "Africa"}, "animals"}, Clauses{"hobby": "golf", "name": "Webbs"}},
+	}
+
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+
+			got := tt.e.Clauses()
+
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
Please remove all usages of `reflect`. You do not need it to complete this, or any, assignment.

Reply:
ok